src/main.rs

```diff
@@ -627,6 +629,50 @@ async fn parse_pull_and_run_settings(matches: &ArgMatches) -> Result<run::PullAn
         .get_one::<bool>("allow-context-aware")
         .unwrap_or(&false)
         .to_owned();
+    if matches.contains_id("record-host-capabilities-interactions")
+        && matches.contains_id("replay-host-capabilities-interactions")
+    {
+        return Err(anyhow!("Cannot use the 'record-host-capabilities-interactions' and the 'replay-host-capabilities-interactions' flags at the same time"));
```

Comment:
Maybe we can fail early in clap. There seems to be a way of defining ArgGroups, that are mutually exclusive.
Example: https://github.com/clap-rs/clap/blob/524e36cf1a67ee6a447d3615a70b065d2b4f5e60/examples/tutorial_derive/04_03_relations.rs#L5-L8 and https://github.com/clap-rs/clap/blob/v3.1.18/examples/tutorial_derive/README.md#argument-relations.

Reply:
Good point, I've implemented that